org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/ScalaOutlinePage.scala

```diff
@@ -0,0 +1,134 @@
+package org.scalaide.ui.internal.editor
+import org.eclipse.ui.views.contentoutline.ContentOutlinePage
+import org.eclipse.swt.widgets.Composite
+import org.eclipse.jface.viewers.ISelectionChangedListener
+import org.eclipse.jface.viewers.SelectionChangedEvent
+import org.eclipse.jface.viewers.ISelection
+import org.eclipse.ui.IActionBars
+import org.eclipse.jdt.internal.ui.actions.CollapseAllAction
+import org.eclipse.ui.handlers.CollapseAllHandler
+import org.eclipse.jface.action.Action
+import org.eclipse.jdt.internal.ui.JavaPluginImages
+import org.eclipse.ui.PlatformUI
+import org.eclipse.jdt.internal.ui.IJavaHelpContextIds
+import org.eclipse.jdt.internal.ui.JavaPlugin
+import org.eclipse.jdt.internal.ui.javaeditor.JavaEditorMessages
+import org.scalaide.logging.HasLogger
+
+class ScalaOutlinePage(val fEditor: ScalaSourceFileEditor) extends ContentOutlinePage with HasLogger {
```

Comment:
The implementation of this class looks like copied from JDT. If so, please add a remark to the Scaladoc that mentions the origin of this code. Also, spend some further time in cleanup. Semi colons should be removed, code should be formatted, members need to be sorted, an empty line between members would be nice.